virtual_rainforest/soil/carbon.py

```diff
@@ -0,0 +1,203 @@
+            soil_temp: soil temperature for each soil grid cell
+            percent_clay: Percentage clay for each soil grid cell
+            dt: time step in a format convertible to minutes
+        """
+        # TODO - Add interactions which involve the three missing carbon pools
+
+        lmwc_from_maom, maom_from_lmwc = self.mineral_association(
+            pH, bulk_density, soil_moisture, soil_temp, percent_clay
+        )
+
+        # Convert dt to a float representing the number of days
+        # 1440 minutes in a day
+        time_step = (dt.astype("timedelta64[m]") / np.timedelta64(1, "m")) / 1440.0
+        print(time_step)
+
+        # Once changes are determined update all pools
+        self.lmwc += lmwc_from_maom * time_step
+        self.maom += maom_from_lmwc * time_step
+
+    def mineral_association(
+        self,
+        pH: NDArray[np.float32],
+        bulk_density: NDArray[np.float32],
+        soil_moisture: NDArray[np.float32],
+        soil_temp: NDArray[np.float32],
+        percent_clay: NDArray[np.float32],
+    ) -> tuple[NDArray[np.float32], NDArray[np.float32]]:
+        """Calculates net rate of LMWC association with soil minerals.
+
+        Following Abramoff et al. (2018), mineral adsorption of carbon is controlled by
+        a Langmuir saturation function. At present, binding affinity and Q_max are
+        recalculated on every function called based on pH, bulk density and clay
+        content. Once a decision has been reached as to how fast pH and bulk density
+        will change (if at all), this calculation may need to be moved elsewhere.
+
+        Args:
+            pH: pH values for each soil grid cell
+            bulk_density: bulk density values for each soil grid cell
+            soil_moisture: soil moisture for each soil grid cell
+            soil_temp: soil temperature for each soil grid cell
+            percent_clay: Percentage clay for each soil grid cell
+        """
+
+        # This expression is drawn from (Mayes et al. (2012))
+        binding_affinity = 10.0 ** (
+            BINDING_WITH_PH["slope"] * pH + BINDING_WITH_PH["intercept"]
+        )
+
+        # This expression is also drawn from Mayes et al. (2012)
+        # Original paper also depends on Fe concentration, but we are ignoring this for
+        # now
+        Q_max = bulk_density * 10 ** (
+            MAX_SORPTION_WITH_CLAY["slope"] * np.log10(percent_clay)
+            + MAX_SORPTION_WITH_CLAY["intercept"]
+        )
+
+        # Using the above calculate the equilibrium MAOM pool
+        equib_maom = (binding_affinity * Q_max * self.lmwc) / (
+            1 + self.lmwc * binding_affinity
+        )
+
+        # Find scalar factors that multiple rates
+        temp_scalar = scalar_temperature(soil_temp)
+        moist_scalar = scalar_moisture(soil_moisture)
+
+        flux = temp_scalar * moist_scalar * self.lmwc * (equib_maom - self.maom) / Q_max
+
+        return -flux, flux
```

Comment:
I'm not sure why you're returning both `flux` and `-flux`. Surely just returning `flux` is enough? The user can work out the other one.

If you do need both and they're always both the same length, then *maybe* this would be better expressed as a 2D numpy array.